go/adbc/driver/bigquery/driver_test.go

```diff
@@ -0,0 +1,1354 @@
+	err = stmt.Bind(ctx, r)
+	if err != nil {
+		return err
+	}
+	err = stmt.SetSqlQuery(insertQuery)
+	if err != nil {
+		return err
+	}
+	rdr, _, err := stmt.ExecuteQuery(ctx)
+	if err != nil {
+		return err
+	}
+
+	rdr.Release()
+	return nil
+}
+
+func (q *BigQueryQuirks) CreateSampleTableWithStreams(tableName string, rdr array.RecordReader) error {
+	var b strings.Builder
+	b.WriteString("CREATE OR REPLACE TABLE ")
+	b.WriteString(q.quoteTblName(tableName))
+	b.WriteString(" (")
+
+	for i := 0; i < rdr.Schema().NumFields(); i++ {
+		if i != 0 {
+			b.WriteString(", ")
+		}
+		f := rdr.Schema().Field(i)
+		b.WriteString(f.Name)
+		b.WriteByte(' ')
+		b.WriteString(getSqlTypeFromArrowField(f))
+	}
+	b.WriteString(")")
+
+	ctx := context.Background()
+	mem := memory.NewCheckedAllocator(memory.DefaultAllocator)
+	tmpDriver := driver.NewDriver(mem)
+	db, err := tmpDriver.NewDatabase(q.DatabaseOptions())
+	if err != nil {
+		panic(err)
+	}
+	defer db.Close()
+
+	cnxn, err := db.Open(ctx)
+	if err != nil {
+		panic(err)
+	}
+	defer cnxn.Close()
+
+	stmt, err := cnxn.NewStatement()
+	if err != nil {
+		panic(err)
+	}
+	defer stmt.Close()
+
+	err = stmt.SetOption(driver.OptionBoolQueryUseLegacySQL, "false")
+	if err != nil {
+		panic(err)
+	}
+
+	creationQuery := b.String()
+	err = stmt.SetSqlQuery(creationQuery)
+	if err != nil {
+		panic(err)
+	}
+	_, err = stmt.ExecuteUpdate(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	// wait for some time before accessing it
+	// BigQuery needs some time to make the table available
+	// otherwise the query will fail with error saying the table cannot be found
+	time.Sleep(5 * time.Second)
+
+	insertQuery := "INSERT INTO " + q.quoteTblName(tableName) + " VALUES ("
+	bindings := strings.Repeat("?,", rdr.Schema().NumFields())
+	insertQuery += bindings[:len(bindings)-1] + ")"
+	err = stmt.BindStream(ctx, rdr)
+	if err != nil {
+		return err
+	}
+	err = stmt.SetSqlQuery(insertQuery)
+	if err != nil {
+		return err
+	}
+	res, _, err := stmt.ExecuteQuery(ctx)
+	if err != nil {
+		return err
+	}
+
+	res.Release()
+	return nil
+}
+
+func (q *BigQueryQuirks) DropTable(cnxn adbc.Connection, tblname string) error {
+	stmt, err := cnxn.NewStatement()
+	if err != nil {
+		return err
+	}
+	defer stmt.Close()
+
+	if err = stmt.SetSqlQuery(`DROP TABLE IF EXISTS ` + q.quoteTblName(tblname)); err != nil {
+		return err
+	}
+
+	_, err = stmt.ExecuteUpdate(context.Background())
+	return err
+}
+
+func (q *BigQueryQuirks) Alloc() memory.Allocator                     { return q.mem }
+func (q *BigQueryQuirks) BindParameter(_ int) string                  { return "?" }
+func (q *BigQueryQuirks) SupportsBulkIngest(string) bool              { return true }
+func (q *BigQueryQuirks) SupportsConcurrentStatements() bool          { return true }
+func (q *BigQueryQuirks) SupportsCurrentCatalogSchema() bool          { return true }
+func (q *BigQueryQuirks) SupportsExecuteSchema() bool                 { return true }
```

Comment:
I don't think this should be `true` either. `ExecuteSchema` currently returns an "unimplemented" error, which should fail this test. I haven't run the tests myself though.